### PR TITLE
Add `run` command

### PR DESCRIPTION
**Pull request by adriantpaez**
Adds the new command `run` to set up and run all the services generated.

## Changes:
- Add `SetupContainers` action
- Add `RunContainers` action
- Add `run` command
- Add missing commands for Windows:
  - `docker compose create`
  - `docker compose build`
- Update commands `docker compose <up,pull,create,build>` to deal with empty services list as an argument, running the action for all the services.
## Types of changes

What types of changes does your code introduce?

- [x] New feature (non-breaking change which adds functionality)

## Testing
**Requires testing**

- [x] Yes

**In case you checked yes, did you write tests?**

- [x] Yes

### CODE REVIEW

File: cli/actions/run.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	GenerationPath string
+	Services       []string
+}
+
+func (s *sedgeActions) RunContainers(options RunContainersOptions) error {
+	upCmd := s.commandRunner.BuildDockerComposeUpCMD(commands.DockerComposeUpOptions{
+		Path:     filepath.Join(options.GenerationPath, configs.DefaultDockerComposeScriptName),
+		Services: options.Services,
+	})
+	_, err := s.commandRunner.RunCMD(upCmd)
```
Comment (AntiD2ta) [inline cli/actions/run.go:20]: Sedge displays every command it runs to the user as a log info. At it very least it displays them as debug, but that's only for not very meaningful commands. The structure will look at this:
```suggestion
	upCmd := s.commandRunner.BuildDockerComposeUpCMD(commands.DockerComposeUpOptions{
		Path:     filepath.Join(options.GenerationPath, configs.DefaultDockerComposeScriptName),
		Services: options.Services,
	})
	log.Infof(configs.RunningCommand, upCmd.Cmd)
	_, err := s.commandRunner.RunCMD(upCmd)
```
DON'T COMMIT THIS SUGGESTION. This is only for illustration purposes, see that the log package is not being imported. I will refer at this change as "Log info the command" or "Log debug the command".

File: cli/actions/setup.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/NethermindEth/sedge/internal/pkg/commands"
+	log "github.com/sirupsen/logrus"
+)
+
+type SetupContainersOptions struct {
+	GenerationPath string
+	Services       []string
+}
+
+func (s *sedgeActions) SetupContainers(options SetupContainersOptions) error {
```
Comment (AntiD2ta) [inline cli/actions/setup.go:16]: Log debug the three commands

File: cli/cli_utils.go [hunk 3]
```diff
@@ -325,6 +325,10 @@ func handleJWTSecret(flags *CliCmdFlags) error {
 		return fmt.Errorf(configs.GenerateJWTSecretError, err)
 	}
 
+	if err := os.MkdirAll(filepath.Dir(flags.jwtPath), os.ModePerm); err != nil {
```
Comment (AntiD2ta) [inline cli/cli_utils.go:328]: Change `os.ModePerm` for `0o755` in this function (the WriteFile was wrong all this time)

**Review by AntiD2ta — CHANGES_REQUESTED** [review]
LGTM, minor changes requested including:

- [x] Update CHANGELOG
- [ ] Add the new command to the commands section in the documentation. Use this [PR](https://github.com/NethermindEth/sedge/pull/204) as reference
- [x] Add License header to all the new files

File: cli/actions/setup.go (hunk 2 above)
Comment (adriantpaez) [reply cli/actions/setup.go:16]: Done 0b0efbfd5e6bc5a5ddaf23a162f0e761610ada2a

File: cli/cli_utils.go (hunk 3 above)
Comment (adriantpaez) [reply cli/cli_utils.go]: Done cf622b0638997f38cfd9274a3331540624bdc38d

File: cli/actions/run.go (hunk 1 above)
Comment (adriantpaez) [reply cli/actions/run.go]: Done https://github.com/NethermindEth/sedge/pull/203/commits/1f1e8620e142d2940d392b8674181bbab4066163

Comment (AntiD2ta) [reply cli/actions/run.go]: Sorry if wasn't clear enough, in this case it is a log.info, as in the suggestion. The reason for that is that this is an important command and the users should be able to tell how to run the compose script in the same way Sedge did.

File: cli/actions/installDeps.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package actions
```
Comment (AntiD2ta) [inline cli/actions/installDeps.go:1]: Add License header please

File: cli/cli_utils.go [hunk 5]
```diff
@@ -325,6 +288,10 @@ func handleJWTSecret(flags *CliCmdFlags) error {
 		return fmt.Errorf(configs.GenerateJWTSecretError, err)
 	}
 
+	if err := os.MkdirAll(filepath.Dir(flags.jwtPath), 0o755); err != nil {
+		return fmt.Errorf(configs.GenerateJWTSecretError, err)
+	}
+
 	err = os.WriteFile(flags.jwtPath, []byte(jwtscret), os.ModePerm)
```
Comment (AntiD2ta) [inline cli/cli_utils.go:295]:
```suggestion
	err = os.WriteFile(flags.jwtPath, []byte(jwtscret), 0o755)
```

**Review by AntiD2ta — COMMENTED** [review]
Some changes still required

File: cli/actions/installDeps.go (hunk 4 above)
Comment (adriantpaez) [reply cli/actions/installDeps.go:16]: Done https://github.com/NethermindEth/sedge/pull/203/commits/255de9066cfa57cdfe92f1dea6850239a09ed565

File: cli/actions/run.go (hunk 1 above)
Comment (adriantpaez) [reply cli/actions/run.go]: Done https://github.com/NethermindEth/sedge/pull/203/commits/bfd0a19c7ac6b66afc1bf73ce4e43711bdc0d821

**Review by AntiD2ta — APPROVED** [review]

**Pull request merged**